http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/AutopauseMonitor.scala

```diff
@@ -25,84 +24,95 @@ class AutopauseMonitor[F[_]](
   jupyterDAO: JupyterDAO[F],
   publisherQueue: Queue[F, LeoPubsubMessage]
 )(implicit
-  F: Async[F],
-  metrics: OpenTelemetryMetrics[F],
-  logger: StructuredLogger[F],
   dbRef: DbReference[F],
   ec: ExecutionContext
-) {
+) extends BackgroundProcess[F, RuntimeToAutoPause] {
+  override def name: String =
+    "autopause" // autopauseRuntime is more accurate. But keep the current name so that existing metrics won't break
+  override def interval: scala.concurrent.duration.FiniteDuration = config.autoFreezeCheckInterval
 
-  val process: Stream[F, Unit] =
-    (Stream.sleep[F](config.autoFreezeCheckInterval) ++ Stream.eval(
-      autoPauseCheck
-        .handleErrorWith(e => logger.error(e)("Unexpected error occurred during auto-pause monitoring"))
-    )).repeat
+  def isAutopauseable(cluster: RuntimeToAutoPause, now: Instant)(implicit
+    F: Async[F],
+    metrics: OpenTelemetryMetrics[F],
+    logger: StructuredLogger[F]
+  ): F[Boolean] =
+    jupyterDAO.isAllKernelsIdle(cluster.cloudContext, cluster.runtimeName).attempt.flatMap {
+      case Left(t) =>
+        logger.error(s"Fail to get kernel status for ${cluster.projectNameString} due to $t").as(true)
```

Comment:
in this case, shall we just log then skip OR we should use the last busyDate? 
I am thinking about the case when Jupyter is stuck in unhealthy state, is that Kernel "busy" or "free".

Reply:
I don't wanna change existing logic in this PR. But the thinking here is if we can't get the status of the kernel, we'll just proceed to pause the runtime